cli/src/commands/help.rs

```diff
@@ -12,38 +12,158 @@
+                    ));
+                } else {
+                    return Err(command_error::user_error(error_str));
+                }
+            }
+        }
+
+        let keyword_list = if ui.color() {
+            format_keywords().ansi().to_string()
+        } else {
+            format_keywords().to_string()
+        };
+
+        write!(ui.stdout(), "{keyword_list}")?;
+
+        return Ok(());
+    } else {
+        let mut args_to_show_help = vec![command.app().get_name()];
+        args_to_show_help.extend(args.command.iter().map(|s| s.as_str()));
+        args_to_show_help.push("--help");
+
+        // TODO: `help log -- -r` will gives an cryptic error, ideally, it should state
+        // that the subcommand `log -r` doesn't exist.
+        let help_err = command
+            .app()
+            .clone()
+            .subcommand_required(true)
+            .try_get_matches_from(args_to_show_help)
+            .expect_err("Clap library should return a DisplayHelp error in this context");
+
+        Err(command_error::cli_error(help_err))
+    }
+}
+
+#[derive(Clone)]
+struct Keyword {
+    name: &'static str,
+    description: &'static str,
+    content: &'static str,
+}
+
+// TODO: Add all documentation to keywords
+//
+// Maybe adding some code to build.rs to find all the docs files and build the
+// `KEYWORDS` at compile time.
+//
+// It would be cool to follow the docs hierarchy somehow.
+//
+// One of the problems would be `config.md`, as it has the same name as a
+// subcommand.
+//
+// TODO: Find a way to render markdown using ANSI escape codes.
+//
+// Maybe we can steal some ideas from https://github.com/martinvonz/jj/pull/3130
+const KEYWORDS: &[Keyword] = &[
+    Keyword {
+        name: "revsets",
+        description: "A functional language for selecting a set of revision",
+        content: include_str!("../../../docs/revsets.md"),
+    },
+    Keyword {
+        name: "tutorial",
+        description: "Show a tutorial to get started with jj",
+        content: include_str!("../../../docs/tutorial.md"),
+    },
+];
+
+fn find_keyword(name: &str) -> Option<&Keyword> {
+    KEYWORDS.iter().find(|keyword| keyword.name == name)
+}
+
+fn format_keywords() -> StyledStr {
+    let keyword_name_max_len = KEYWORDS
+        .iter()
+        .map(|keyword| keyword.name.len())
+        .max()
+        .unwrap();
+
+    let mut ret = String::new();
+
+    writeln!(ret, "{}", "Help Keywords:".bold().underlined()).unwrap();
+    for keyword in KEYWORDS {
+        write!(
+            ret,
+            "  {}  ",
+            format!("{:<keyword_name_max_len$}", keyword.name).bold()
+        )
+        .unwrap();
+        writeln!(ret, "{}", keyword.description).unwrap();
+    }
+
+    ret.into()
+}
+
+pub fn format_keywords_for_after_help() -> StyledStr {
+    let mut ret = StyledStr::new();
+    writeln!(
+        ret,
+        "Use {} to show help related to these keywords",
+        "'jj help -k'".bold()
+    )
```

Comment:
Since we have a flag to show the list, maybe we can remove it from the top-level help? I think "Use jj help -k ..." is good enough for discoverability.

Then, we can use our `ui.formatter()` API to print the list of keywords.